openerp/addons/base/tests/test_orm.py

```diff
@@ -166,6 +166,22 @@ def read_group(interval, domain=None):
                         ['date'], ['date:month', 'date:day'], lazy=False)
         self.assertEqual(len(rg), len(all_partners))
 
+    def test_m2m_store_trigger(self):
+        group_user = self.env.ref('base.group_user')
+
+        user = self.env['res.users'].create({
+            'name': 'test',
+            'login': 'test_m2m_store_trigger',
+            'groups_id': [(6, 0, [])],
+        })
+        self.assertTrue(user.share)
+
+        group_user.write({'users': [(4, user.id)]})
+        self.assertFalse(user.share)
```

Comment:
So that works through new api magic? Because the [many2many setter](https://github.com/odoo/odoo/blob/bcc136f3e51ecdd54c7e28cee364b64aefa96ffe/openerp/osv/fields.py#L1034-L1070) never writes on the comodel of a m2m for LINK(4)/UNLINK(3) operations.